<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="description" content="">
    <title>Pantograph - Pangenome Browser</title>

    <!-- Bootstrap core CSS -->
    <link href="https://getbootstrap.com/docs/4.4/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">

    <!-- Custom styles for this template -->
    <link href="style.css" rel="stylesheet">
</head>

<body>

    <header>
        <img src="img/pantograph.png" class="titles">
    </header>

    <div class="container">
        <div class="row">
            <div class="col-md-4">
                <h2>Online Browser Demo </h2>
                <p>
                    Browse our proud visualization of a subset of the COVID-19 graphical pangenome from 
                    the <a href="https://workbench.lugli.arvadosapi.com/projects/lugli-j7d0g-825x3r5vcs41dus">Public Sequence Resource</a>.
                    Thanks to the hard work of the <a href="https://github.com/arvados/bh20-seq-resource/graphs/contributors">PubSeq team</a>, 
                    we are able visualize 169 aligned individuals using Pantograph.
                </p>
                <p><a class="btn btn-secondary" href="https://graph-genome.github.io//Schematize/" role="button">Pangenome Browser &raquo;</a></p>
            </div>
            <div class="col-md-4">
                <h2>Learn About Pantograph</h2>
                <p>
                    Start by reading about the pangenome mapping which shows variations
                    of a specific genome.
                </p>

                <p><a class="btn btn-secondary" href="pantograph.html" role="button">Learn About Pantograph &raquo;</a></p>
            </div>
            <div class="col-md-4">
                <h2>Get Involved</h2>
                <p>
                    Start by reading the high level documentation. Then select an
                    issue in the issues database to work on. Your efforts are greatly
                    appreciated and may help to save many lives.
                </p>
                <p><a class="btn btn-secondary" href="getinvolved.html" role="button">Get Involved &raquo;</a></p>
            </div>
        </div>

        <a href="project.html">Project Documents</a>

        <hr>

    </div> <!-- /container -->

    <main role="main">
        <div class="card">
            <div class="card-header">
                Software Developers and Project Managers: You can help us outmaneuver COVID-19!
            </div>

            <div class="card-body">


                <p><strong>Problem:</strong> As COVID-19 spreads, it can mutate. Mutations make vaccines and tests ineffective.
                </p>

                <p><strong>Solution:</strong> Pantograph—a War Room Tool for biologists to track mutations. Real-time visualization
                    empowers us to stay ahead of the fight with new vaccines, tools, and tests.</p>

                <p><strong>Method:</strong> Open Source project on GitHub. Features will be broken down into issues.</p>

                <p><strong>Goal:</strong> Open Source data collection point for: </p>
                <ul>
                    <li>Health care providers to note mutations on Pantograph</li>
                    <li>Biologists to immediately flag mutations on a global map and track new strains</li>
                    <li>Vaccine developers to stay ahead of mutations with new vaccines</li>
                </ul>

                <p><strong>What we need:</strong> Software Developers (no biology background necessary) and Project Managers.
                    10-20 hours/week volunteer time to develop this crucial tool.
                    Most beneficial <a href="hackathon.html#skills-needed">skills</a>:
                    JavaScript, React, Semantic Web using SPARQL</p>

                <p><strong>Software Developers:</strong></p>
                <ul>
                    <li>You’ll work on an issue and will make a pull request</li>
                    <li>Each team’s Features will be merged into a larger project</li>
                    <li>We’ll create a powerful War Room Tool together</li>
                </ul>

                <p><strong>Project Managers:</strong></p>
                <ul>
                    <li>You’ll work on a Project and will monitor task progress</li>
                    <li>You’ll manage communication between developers</li>
                    <li>You’ll oversee testing codes, finding bugs, and posting issues</li>
                </ul>


                <p><strong>Join us:</strong> Drop a mail to josiah@newline.us and
                    <a href="getinvolved.html">get involved</a>!
                </p>

                <p>
                    <a class="btn btn-primary btn-lg" href="project.html" role="button">Pantograph Pangenome Browser &raquo;</a>
                </p>


                <div class="card">
                    <div class="card-body">
                        <h5>The Pantograph Team</h5>
                        <ul>
                            <li>Josiah Seaman, Kew Gardens London</li>
                            <li>Simon Heumos, <a href="https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/">QBiC</a></li>
                            <li>Erik Garrison, University of California Santa Clara</li>
                            <li>Thomas Townsley, University of Lipscomb</li>
                            <li>Andrea Guarracino, Centre for Molecular Bioinformatics, Department of Biology, University Of Rome Tor Vergata, Rome, Italy</li>
                            <li><a href="https://www.bonfacemunyoki.com/bio/">Bonface Munyoki</a></li>
                            <li>Dmytro Trybushni, Karlsruhe Institute Of Technology, Germany</li>
                            <li>Mark Seaman, University of Northern Colorado</li>
                            <li>Njagi Mwaniki, IDEAL fellow at KEMRI-Wellcome Trust, Pwani University</li>
                            <li>Jörg Hagmann, <a href="https://computomics.com">Computomics GmbH</a></li>
                            <li>Stacie Seaman, Northern Colorado Medical Center</li>
                            <li>Toshiyuki Yokoyama, University of Tokyo</li>
                            <li>Torsten Pook, University Göttingen</li>
                            <li>Christian Kubica, Max Planck Institute for Developmental Biology, Tübingen</li>
                            <li>Jerven Bolleman, Swiss Institute of Bioinformatics, Geneva</li>
                            <li>Artem Tarasov</li>
                            <li>Hannah Sewell, InPart Inc.</li>
                            <li>Eloi Durant</li>
                            <li>Kaytie Innamorati</li>
                            <li>Michael Crusoe</li>
                            <li>Peter Amstutz</li>
                            <li>Robert Fornof</li>
                            <li>Travis Clark</li>
                            <li>Jack Tierney</li>
                            <li>Jacob Windsor</li>
                        </ul>


                        <p><strong>The Pantograph project is supported by</strong>
                            <p>
                                <img src="img/ZIM.jpg">
                            </p>


                            <p>
                                <img src="img/GermanNetworkForBioInformatics1.jpg" width="400">
                            </p>

                            <p>
                                <img src="img/BBSRC.png" width="400">
                            </p>


                            <p>
                                The <a href="https://www.denbi.de/cloud">deNBI Cloud</a> has provided VMs for developing this project and is currently providing a VM of 28 cores, 64GB RAM and 1TB storage for our <strong>COVID19 Pangenome Visualization</strong>.
                                The deNBI cloud <a href="https://cloud.denbi.de/covid-19-resources/">covid-19-resources</a> lists us as one of the available resources to fight COVID19.
                            </p>


                            <p>Pangenome Schematic design and development research was funded through the 1001 Genomes project under grant BBSRC BB/S004661/1.
                            </p>


                            <p>NBDC/DBCLS BioHackathon 2019 is supported by the Integrated Database Project (Ministry of Education, Culture, Sports, Science and Technology of Japan) and hosted by the National Bioscience Database Center (NBDC) and the Database Center for Life Science (DBCLS).
                            </p>


                            <p>Computomics GmbH funded a second hackathon to implement the data pipeline in November 2019.
                            </p>


                            <p>Toshiyuki Yokoyama’s work was funded in part by JSPS KAKENHI (grant numbers 16H06279 and 19J21608).
                            </p>


                            <p>If you wish to donate to the Pantograph project, e.g. for
                                infrastructure like setting up and maintaining a public server:
                            </p>
                            <p>
                                <a class="btn btn-primary btn-lg" href="https://paypal.me/Computomics" rel="noopener noreferrer" target="_blank" role="button">Donate to Pantograph Project &raquo;</a>
                            </p>

                            <p>If your company or organization would like to get more involved in developing this tool,
                                please contact us: <a href="mailto: pantograph@computomics.com">pantograph@computomics.com</a>.
                            </p>
                    </div>
                </div>
            </div>
        </div>
    </main>


    <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
    <script>
        window.jQuery || document.write('<script src="/docs/4.4/assets/js/vendor/jquery.slim.min.js"><\/script>')

    </script>
    <script src="/docs/4.4/dist/js/bootstrap.bundle.min.js" integrity="sha384-6khuM
